File: gateway-service/app/service/UserService.js
const userAPI = require("../api/UserAPI");

class UserService {
  async getUserBalance(userId) {
    return await userAPI.getBalanceUser(userId);
  }

  async getUser(userId) {
    return await userAPI.getUser(userId);
  }
}

module.exports = new UserService();

File: account-service/src/main/java/com/banvien/training/service/UserService.java
package com.banvien.training.service;

import com.banvien.training.dto.response.StatusPaymentResponse;
import com.banvien.training.dto.response.UserBalanceResponse;
import com.banvien.training.dto.response.UserResponse;

public interface UserService {
    UserBalanceResponse getUserBalance(Long userId);

    UserResponse getUser(Long userId);

    StatusPaymentResponse deductMoney(Long userId, Long amount);

    StatusPaymentResponse refundMoney(Long userId, Long amount);
}
